tests/unit/amo/components/TestAddonsByAuthorsCard.js

```diff
@@ -346,6 +344,26 @@ describe(__filename, () => {
     sinon.assert.notCalled(dispatchSpy);
   });
 
+  it('should have extensions length equal to numberOfAddons props', () => {
```

Comment:
We don't need these tests, my apologies if my earlier comments suggested this is what was needed. The `numberOfAddons` passed into `AddonsByAuthorsCard` is only used by the `AddonsCard` to determine the number of placeholders, which isn't what's being tested here.

The test that would be useful would be a test of the `UserProfile` component to check that the properties passed into `AddonsByAuthorsCard` is correct for both the card rendered for extensions and the card rendered for themes.

Reply:
@bobsilverberg , following are my points.
1. `numberOfAddons` is not just for placeholders but also for number of addons in the card, have a look here (https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/AddonsByAuthorsCard/index.js#L194), we use `numberOfAddons` to slice out addons.
2. `EXTENSIONS_BY_AUTHORS_PAGE_SIZE` and `THEMES_BY_AUTHORS_PAGE_SIZE` are used for maximum number of addons that will be fetched from API, and then in component based on `numberOfAddons` we show addons.
3. My initial PR was failed because I used `12` in `numberOfAddons` key but didn't change in reducer which was `6`, the testcases written are covering that case for both theme and extension, for example if I put 14 as `numberOfAddons` test value the test should fail as its getting only `12` from reducer.

Reply:
> 1. `numberOfAddons` is not just for placeholders but also for number of addons in the card, have a look here (https://github.com/mozilla/addons-frontend/blob/master/src/amo/components/AddonsByAuthorsCard/index.js#L194), we use `numberOfAddons` to slice out addons.

Ah, yes, I missed the code in `mapStateToProps`. You are correct, so this is good.

> 2. `EXTENSIONS_BY_AUTHORS_PAGE_SIZE` and `THEMES_BY_AUTHORS_PAGE_SIZE` are used for maximum number of addons that will be fetched from API, and then in component based on `numberOfAddons` we show addons.

Ok, given that it seems like the tests should actually request a `numberOfAddons` that is less than the corresponding constant (e.g., `EXTENSIONS_BY_AUTHORS_PAGE_SIZE`) to that we can prove that the list of add-ons rendered is in fact controlled by `numberOfAddons`. Does that make sense?

Reply:
> Ok, given that, it seems like the tests should actually request a numberOfAddons that is less than the corresponding constant (e.g., EXTENSIONS_BY_AUTHORS_PAGE_SIZE) so that we can prove that the list of add-ons rendered is in fact controlled by numberOfAddons. Does that make sense?

Sure that is good, I will update, also there are some conflicts now I will fix up